docs/core/entity.md

```diff
@@ -119,6 +119,17 @@ The following properties are also available on entities. However, they are for a
 | state_attributes                | <code>dict &#124; None</code> | `None` | State attributes of a base domain. This property is implemented by the domain base entity and should not be implemented by integrations.
 | unit_of_measurement             | <code>str &#124; None</code> | The unit of measurement that the entity's state is expressed in. In most cases, for example for the `number` and `sensor` domains, this is implemented by the domain base entity and should not be implemented by integrations.
 
+## State attributes
+
+Entities can provide additional state attributes besides the built-in ones, to provide further details to its state.
+This should be done by providing a dictionary with keys and values to the `extra_state_attributes` property.
+
+Providing state attributes comes with the following rules:
+
+- If an attribute is expected, it should be in the dictionary. Attributes should not "come and go".
+- If an attribute is expected, but it's not providing a value right now, its value should be `None` and remain in the dictionary.
+- If an attribute is not expected, it should not be in the dictionary. An attribute is expected if supported by the entity, for example as indicated by `supported_features`.
```

Comment:
I am not really sure what this means and how it would look in practice

Reply:
As an example, in `statistics` the `buffer_usage_ratio` attribute is only shown if the sensor is configured to handle a buffer size. Otherwise this attribute would be irrelevant hence not showing in that case.

Reply:
This is the developer docs not the user docs. The attribute is expected if it's part of a supported feature and that feature is supported or if it's an extra state attribute the attribute is expected if the device has support for that attribute.